### PR TITLE
feat: handle mixed protocol for conversation operations - WPB-529

### PR DESCRIPTION
**Pull request by netbe**
<table>
<td>
  <a href="https://wearezeta.atlassian.net/browse/WPB-529" title="WPB-529" target="_blank">StoryWPB-529</a>  [iOS] Maintaining MLS group's clients list during migration
  </td></table>
  <br />
 

----
#### PR Submission Checklist for internal contributors

- The **PR Title**
  - [ ] conforms to the style of semantic commits messages¹ supported in Wire's Github Workflow²
  - [ ] contains a reference JIRA issue number like `SQPIT-764`
  - [ ] answers the question: _If merged, this PR will: ..._ ³

- The **PR Description**
  - [ ] is free of optional paragraphs and you have filled the relevant parts to the best of your ability
----

# What's new in this PR?

Updates normal MLS group operations (adds, removes, updates etc) for conversations in mixed mode
#### Test Coverage (Optional)

- [x] I have added automated test to this contribution

----
#### PR Post Submission Checklist for internal contributors (Optional)

 - [ ] Wire's Github Workflow has automatically linked the PR to a JIRA issue
----
#### PR Post Merge Checklist for internal contributors

 - [ ] If any soft of configuration variable was introduced by this PR, it has been added to the relevant documents and the CI jobs have been updated.
----
##### References
1. https://sparkbox.com/foundry/semantic_commit_messages
1. https://github.com/wireapp/.github#usage
1. E.g. `feat(conversation-list): Sort conversations by most emojis in the title #SQPIT-764`.

### CODE REVIEW

File: wire-ios-request-strategy/Sources/Request Strategies/Conversation/ConversationEventProcessorTests.swift [hunk 1]
```diff
@@ -115,7 +115,19 @@ class ConversationEventProcessorTests: MessagingTestBase {
...
+
+    func test_UpdateConversationMemberLeave_WipesMLSGroup_WithProtocolMixed() {
+        internalTest_UpdateConversationMemberLeave_WipesMLSGroup(messagingProtocol: .mixed, shouldWipe: true)
+    }
+
+    func test_UpdateConversationMemberLeave_DoesntWipeMLSGroup_WhenProtocolIsNotMLS() {
+        internalTest_UpdateConversationMemberLeave_WipesMLSGroup(messagingProtocol: .proteus, shouldWipe: false)
+    }
+
+    func internalTest_UpdateConversationMemberLeave_WipesMLSGroup(messagingProtocol: MessageProtocol, shouldWipe: Bool) {
```
Comment (David-Henner) [inline wire-ios-request-strategy/Sources/Request Strategies/Conversation/ConversationEventProcessorTests.swift:130]: You could do that ⬇️ and add the file and line to the assert
```suggestion
    func internalTest_UpdateConversationMemberLeave_WipesMLSGroup(messagingProtocol: MessageProtocol, shouldWipe: Bool, file: StaticString = #file, line: UInt = #line) {
```

File: wire-ios-sync-engine/Source/UserSession/URLActionProcessors/DeepLinkURLActionProcessor.swift [hunk 2]
```diff
@@ -145,7 +145,10 @@ class DeepLinkURLActionProcessor: URLActionProcessor {
                 return
             }
 
-            guard let groupId = upToDateConversation.mlsGroupID, upToDateConversation.messageProtocol == .mls else {
+            guard
```
Comment (David-Henner) [inline wire-ios-sync-engine/Source/UserSession/URLActionProcessors/DeepLinkURLActionProcessor.swift:148]: swiflint error here

File: wire-ios-sync-engine/Tests/Source/Integration/ConnectionTests.m [hunk 3]
```diff
@@ -612,7 +613,8 @@ - (void)DISABLED_testThatWeDontSeeASystemMessageWhenAUserAcceptsAConnectionReque
         XCTAssertEqual(conv1.conversationType, ZMConversationTypeOneOnOne);
         XCTAssertEqual(conv1.allMessages.count, 1u, @"%@", [conv1 lastMessagesWithLimit:50]);
         ZMSystemMessage *message1 = (ZMSystemMessage *)conv1.lastMessage;
-        XCTAssertEqual(message1.systemMessageType, ZMSystemMessageTypeConnectionRequest);
+        // removed as it does not compile because of deprecation.
+//        XCTAssertEqual(message1.systemMessageType, ZMSystemMessageTypeConnectionRequest);
```
Comment (David-Henner) [inline wire-ios-sync-engine/Tests/Source/Integration/ConnectionTests.m:617]: is this meant to be here?

Comment (netbe) [reply wire-ios-sync-engine/Tests/Source/Integration/ConnectionTests.m:617]: yes as we treat warnings as errors, it was not compiling, but as you see it is a disabled test. See above TODO

**Comment by David-Henner** [comment]
.

**Comment by David-Henner** [comment]
.

**Pull request closed without merging**